python/feathub/processors/spark/dataframe_builder/over_window_utils.py

```diff
@@ -0,0 +1,207 @@
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from datetime import timedelta
+from typing import Optional, Sequence, List, Dict
+
+from feathub.common.exceptions import FeathubTransformationException, FeathubException
+from feathub.feature_views.transforms.agg_func import AggFunc
+from feathub.feature_views.transforms.over_window_transform import OverWindowTransform
+from feathub.processors.spark.dataframe_builder.spark_dataframe_builder_constants import (  # noqa
+    TIME_ATTRIBUTE_NAME,
+)
+from feathub.processors.spark.dataframe_builder.spark_sql_expr_utils import (
+    to_spark_sql_expr,
+)
+from feathub.processors.spark.dataframe_builder.aggregation_utils import (
+    AggregationFieldDescriptor,
+)
+from pyspark.sql.window import Window
+from pyspark.sql import DataFrame, WindowSpec, Column, functions
+
+
+class OverWindowDescriptor:
+    """
+    Descriptor of an over window.
+    """
+
+    def __init__(
+        self,
+        window_size: Optional[timedelta],
+        limit: Optional[int],
+        group_by_keys: Sequence[str],
+        filter_expr: Optional[str],
+    ) -> None:
+        self.window_size = window_size
+        self.limit = limit
+        self.group_by_keys = group_by_keys
+        self.filter_expr = filter_expr
+
+    @staticmethod
+    def from_over_window_transform(
+        window_agg_transform: OverWindowTransform,
+    ) -> "OverWindowDescriptor":
+        filter_expr = (
+            to_spark_sql_expr(window_agg_transform.filter_expr)
+            if window_agg_transform.filter_expr is not None
+            else None
+        )
+        return OverWindowDescriptor(
+            window_agg_transform.window_size,
+            window_agg_transform.limit,
+            window_agg_transform.group_by_keys,
+            filter_expr,
+        )
+
+    def __eq__(self, other: object) -> bool:
+        return (
+            isinstance(other, self.__class__)
+            and self.window_size == other.window_size
+            and self.limit == other.limit
+            and self.group_by_keys == other.group_by_keys
+            and self.filter_expr == other.filter_expr
+        )
+
+    def __hash__(self) -> int:
+        return hash(
+            (self.window_size, self.limit, tuple(self.group_by_keys), self.filter_expr)
+        )
+
+
+def _get_over_window_agg_column_list(
+    window_descriptor: "OverWindowDescriptor",
+    agg_descriptors: List["AggregationFieldDescriptor"],
+    window_spec: WindowSpec,
+) -> Dict[str, Column]:
+    return {
+        descriptor.field_name: _get_over_window_agg_select_expr(
```

Comment:
Should we rename `_get_over_window_agg_select_expr` as `_get_over_window_agg_column` for consistency with `_get_over_window_agg_column_list`?